app/javascript/components/admin/AuditModal.js

```diff
@@ -200,7 +200,8 @@ class AuditModal extends React.Component {
         } else {
           return (
             <span>
-              <b>Role</b>: Changed from &quot;{change.details[0]}&quot; to &quot;{change.details[1]}&quot;
+              <b>Role</b>: Changed from &quot;{_.map(change.details[0].split('_'), _.capitalize).join(' ')}&quot; to &quot;
```

Comment:
So actually to avoid this duplicate logic (since this is already done [here](https://github.com/SaraAlert/SaraAlert/blob/master/app/views/admin/index.html.erb#L4)) you can just pass the `role_types` prop to the `AuditModal` component from the `AdminTable` component and use that here.

Reply:
Looking more closely I don't think there's an easy way to use the role_types prop here: because role_types is just an array of the correctly formatted values, there's no way to associate the particular roles we need when we render this component with the values outside it. This seems like a common enough operation to probably warrant a utilities function but I don't immediately see a great place to put one.

Reply:
You're right, forgot that here `prop_types` was an array rather than a hash. At the very least can you pull the code out here into a small method in this file that you can call twice here instead?
Something like:
```
splitAndCapitalize(str) {
    return _.map(str.split('_'), _.capitalize).join(' ');
}
```

Reply:
Why not use `_.startCase(change.details[0])`? https://lodash.com/docs/4.17.15#startCase

Reply:
Even better!

Reply:
Done. This also seems like a common enough operation to warrant being put in a generic string formatting utilities file at some point

Reply:
@ashankland I like @holmesie 's suggestion of the existing Lodash function mentioned above. Cleaner and doesn't require an extracted function.